http-server-netty/src/main/java/io/micronaut/http/server/netty/MicronautHttpData.java

```diff
@@ -134,7 +129,7 @@ private Chunk lastChunk() {
      * @return The chunk, or {@code null} if this data is {@link #isCompleted() completed} and all
      * chunks have been polled.
      */
-    public Chunk pollChunk() {
+    public synchronized Chunk pollChunk() {
```

Comment:
these syncs should not be necessary. These calls should not happen concurrently

Reply:
Reverted